### DISCUSSION TITLE
Change ID to UUID

### DISCUSSION THREAD

**Discussion by Lucas-Gruber** [directus/directus#7839] [Archive]
Hey, I have a question, is it possible to change the ID of a data model from a number to a UUID without recreating everything? Whether in the app or in the database, the main thing is that it works.

Thank you.

**Comment by benhaynes** [comment]
It's not something Directus will automate. That said, you can go into the Database to do this:

1. Update the ID field type to UUID (disable FK constraints as needed), and any FK fields
2. Create a script that gives each item a UUID, and immediately updates all relational records to the matching key
3. Go into Directus and make sure the config there matches
4. Re-enable FK constraints

Should be all set.

This is the type of thing that automation would be great for, and we'll (eventually) add this into Directus (or an extension), but for now this is the best route.
